app/Models/Farmasi/ResepObat.php

```diff
@@ -113,23 +132,11 @@ public function scopeKunjunganResep(Builder $query, string $jenisResep = 'umum',
             ->join('penjab', 'reg_periksa.kd_pj', '=', 'penjab.kd_pj')
             ->join('dokter', 'resep_obat.kd_dokter', '=', 'dokter.kd_dokter')
             ->join('poliklinik', 'reg_periksa.kd_poli', '=', 'poliklinik.kd_poli')
-            ->whereBetween('resep_obat.tgl_perawatan', [$tglAwal, $tglAkhir])
-            ->where('resep_obat.tgl_perawatan', '>', '0000-00-00')
-            ->when($shift === 'Pagi', function ($q) {
-                $q->whereBetween('resep_obat.jam', ['07:00:01', '14:00:00']);
+            ->where(function ($q) use ($waktuAwal, $waktuAkhir, $waktuAwalAkhir, $waktuAkhirAkhir) {
```

Comment:
```suggestion
            ->where(function ($q) use ($tglAwal, $tglAkhir, $waktuShift, $shift) {
```

dibagian ini sudah di ubah seperti ini